bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -452,8 +454,8 @@ def to_api_repr(self):
         :returns: JSON mapping
         """
         types = [
-            {'name': name, 'type': self.struct_types[name]}
-            for name in self._order
+            {'name': key, 'type': value}
+            for key, value in self.struct_types.items()
```

Comment:
I encourage the use of `six.iteritems()`, though not sure what the expected size of items is to know if a copy is prohibitive